cmd/playground/interactive_window.go

```diff
@@ -0,0 +1,213 @@
+}
+
+// this handles motion of the pointer when we hit arrow-up key
+func (m *model) handleUp() {
+	switch m.cursor {
+	case "Offsetinput":
+		//here we are shifting from Offsetinput to Sizeinput so we have to stop updating the Offsetinput(hence Offset.Blur())
+		// and start updating Sizeinput(hence Sizeinput.Focus())
+		m.Offsetinput.Blur()
+		m.Sizeinput.Focus()
+		m.cursor = "Sizeinput"
+	case "TruncateTo":
+		m.Offsetinput.Focus()
+		m.cursor = "Offsetinput"
+	case "year", "month", "day", "hour", "minute":
+		m.cursor = "TruncateTo"
+	}
+}
+
+func (m *model) handleDown() {
+	switch m.cursor {
+	case "TruncateTo":
+		m.cursor = "hour"
+	case "Offsetinput":
+		m.Offsetinput.Blur()
+		m.cursor = "TruncateTo"
+	case "Sizeinput":
+		m.Offsetinput.Focus()
+		m.Sizeinput.Blur()
+		m.cursor = "Offsetinput"
+	}
+}
+
+// handles left and right arrow key movements
+func (m *model) handleRight() {
+	switch m.cursor {
+	case "month":
+		m.cursor = "year"
+	case "day":
+		m.cursor = "month"
+	case "minute":
+		m.cursor = "day"
+	case "hour":
+		m.cursor = "minute"
+	}
+}
+
+func (m *model) handleLeft() {
+	switch m.cursor {
+	case "minute":
+		m.cursor = "hour"
+	case "day":
+		m.cursor = "minute"
+	case "month":
+		m.cursor = "day"
+	case "year":
+		m.cursor = "month"
+	}
+}
+
+// update the state of the system when a msg(Key strokes , mouse clicks ..) is given
+func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	// respond to only key strokes
+	case tea.KeyMsg:
+		switch msg.String() {
+		// to terminate the process
+		case "ctrl+c", "q":
+			return m, tea.Quit
+		case "up":
+			m.handleUp()
+		case "down":
+			m.handleDown()
+		case "left":
+			m.handleLeft()
+		case "right":
+			m.handleRight()
+		// increase or decrease the value of the parameter the cursor points at
+		case "shift+up", "shift+right":
+			switch m.cursor {
+			case "TruncateTo":
+				m.state.windowv2.TruncateTo = m.state.IncrementTruncate()
+			case "year", "month", "day", "hour", "minute":
+				m.state.IncrementTime(m.cursor)
+			}
+		case "shift+down", "shift+left":
+			switch m.cursor {
+			case "TruncateTo":
+				m.state.windowv2.TruncateTo = m.state.DecrementTruncate()
+			case "year", "month", "day", "hour", "minute":
+				m.state.DecrementTime(m.cursor)
+			}
+		// if not the arrow keys then the input must be given to the size or the off set hence we update them accordingly
+		default:
+			// to get updated the input fields must be focussed since we handle the Focus and Blur we can directly update them here
+			m.Sizeinput, _ = m.Sizeinput.Update(msg)
+			m.Offsetinput, _ = m.Offsetinput.Update(msg)
+		}
+	}
+	// update the values of the window versions in state for every modification from input
+	m.updateWindowVersions()
+	return m, nil
+}
+
+// we genarate a cursor if the value of the cursor matches with the current field name, else we pass a blank instead
+func (m model) genarateFields(position string, value string) string {
```

Comment:
try to change this method to better reflect what it does. currently I'm not sure what this method does. if what you mean is to generate input fields view, then perhaps you can name it like `generateInputFieldsView` or other better name. With this name, I first assumed we will get a list of fields.